test/extensions/filters/http/adaptive_concurrency/concurrency_controller/gradient_controller_test.cc

```diff
@@ -488,6 +488,8 @@ TEST_F(GradientControllerTest, TimerAccuracyTest) {
   // random value > 10% of the interval.
   EXPECT_CALL(random_, random()).WillOnce(Return(15000));
   EXPECT_CALL(*rtt_timer, enableTimer(std::chrono::milliseconds(105000), _));
+  // Verify the sample timer is reset after the minRTT calculation occurs.
+  EXPECT_CALL(*sample_timer, enableTimer(std::chrono::milliseconds(123), _));
```

Comment:
Can you remove the NiceMock from both of these timers in each test? This would have caught this bug more readily.